helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/client/HelpDeskPublicApiClient.java

```diff
@@ -0,0 +1,118 @@
+package org.symphonyoss.symphony.bots.helpdesk.bot.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+import org.symphonyoss.symphony.bots.helpdesk.bot.config.HelpDeskBotConfig;
+import org.symphonyoss.symphony.pod.model.CompanyCert;
+import org.symphonyoss.symphony.pod.model.CompanyCertDetail;
+
+import java.io.IOException;
+import java.net.HttpCookie;
+import java.util.List;
+
+/**
+ * Provides acess to public APIs of a POD.
+ * Created by campidelli on 03/12/17.
+ */
+@Service
+public class HelpDeskPublicApiClient {
+
+  @Autowired
+  HelpDeskBotConfig config;
+
+  private static final String GET_SALT_ENDPOINT = "login/salt?userName=%s";
+
+  private static final String LOGIN_ENDPOINT = "login/username_password";
+
+  private static final String CREATE_CERTIFICATE_ENDPOINT = "pod/v2/companycert/create";
+
+  private static final String SESSION_KEY = "skey";
+
+  private RestTemplate restTemplate = new RestTemplate();
+
+  /**
+   * Gets a generated salt for a given userName.
+   * @param userName User name.
+   * @return Generated salt.
+   */
+  public String getSalt(String userName) {
+    String endpoint = config.getPod().getUrl(GET_SALT_ENDPOINT);
+    endpoint = String.format(endpoint, userName);
+
+    ResponseEntity<String> response = restTemplate.getForEntity(endpoint, String.class);
+
+    String json = response.getBody();
+    ObjectMapper mapper = new ObjectMapper();
```

Comment:
The ObjectMapper object is too expensive to be created every time you call this method. Please move it as constant field and perform an eager initialization.

Reply:
Done.